src/Features/Core/Portable/Completion/Providers/AbstractAwaitCompletionProvider.cs

```diff
@@ -43,41 +65,67 @@ public sealed override async Task ProvideCompletionsAsync(CompletionContext cont
             var semanticModel = await document.ReuseExistingSpeculativeModelAsync(position, cancellationToken).ConfigureAwait(false);
             var workspace = document.Project.Solution.Workspace;
             var syntaxContext = document.GetRequiredLanguageService<ISyntaxContextService>().CreateContext(workspace, semanticModel, position, cancellationToken);
-            if (!syntaxContext.IsAwaitKeywordContext())
+
+            var isAwaitKeywordContext = syntaxContext.IsAwaitKeywordContext();
+            var isDotAwaitContext = IsDotAwaitKeywordContext(syntaxContext, cancellationToken);
+            Debug.Assert(!(isAwaitKeywordContext && isDotAwaitContext), "isDotAwaitContext and isAwaitKeywordContext should never be both true.");
+            if (!isAwaitKeywordContext && !isDotAwaitContext)
             {
                 return;
             }
 
             var generator = SyntaxGenerator.GetGenerator(document);
             var syntaxKinds = document.GetRequiredLanguageService<ISyntaxKindsService>();
-            var completionItem = GetCompletionItem(syntaxContext.TargetToken, generator, syntaxKinds, syntaxFacts);
+            var completionItem = GetCompletionItem(syntaxContext.TargetToken, isAwaitKeywordContext, isDotAwaitContext, generator, syntaxKinds, syntaxFacts);
             context.AddItem(completionItem);
         }
 
         public sealed override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, char? commitKey = null, CancellationToken cancellationToken = default)
         {
-            // IsComplexTextEdit is true when we want to add async to the container.
+            // IsComplexTextEdit is true when we want to add async to the container or place await in front of the expression.
             if (!item.IsComplexTextEdit)
             {
                 return await base.GetChangeAsync(document, item, commitKey, cancellationToken).ConfigureAwait(false);
             }
+            var completionChangeEdit = (CompletionChangeEdit)Enum.Parse(typeof(CompletionChangeEdit), item.Properties[nameof(CompletionChangeEdit)]);
+            using var _ = ArrayBuilder<TextChange>.GetInstance(out var builder);
+            var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var syntaxKinds = document.GetRequiredLanguageService<ISyntaxKindsService>();
```

Comment:
```suggestion
            var syntaxKinds = syntaxFacts.SyntaxKinds;
```

(up to you if you want to change this)